libmcount/mcount-dynamic.c

```diff
@@ -0,0 +1,91 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <assert.h>
+#include <signal.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/uio.h>
+#include <sys/ptrace.h>
+#include <signal.h>
+
+#define PR_FMT     "mcount-dynamic"
+#define PR_DOMAIN  DBG_MCOUNT
+
+#include "libmcount/mcount.h"
+#include "libmcount/internal.h"
+#include "utils/utils.h"
+#include "utils/symbol.h"
+#include "utils/filter.h"
+#include "utils/script.h"
+#include "utils/env-file.h"
+
+/*
+ * previous mcount_startup() the contructor.
+ */
+void pre_startup(void)
+{
+	int fd_envfile;
+
+	fd_envfile = open_env_file();
+	set_env_from_file(fd_envfile);
```

Comment:
After `pre_startup()` does `setenv()`, `mcount_startup()` does `getenv()` to pass various uftrace option values.
But I think we can skip `setenv()` and `getenv()`.
To sum up, after reading "envfile", we can directly set uftrace option values. What do you think about this ?

Reply:
@Taeung you're right, but I'm considering that dynamic tracing feature need more time to stable. for that reason, I want to split dynamic tracing feature from uftrace. it is the reason what I have wrote code this way. if you have more idea about isolate the dynamic tracing feature, let me know please.